src/System/Taffybar/Widget/Brightness.hs

```diff
@@ -0,0 +1,13 @@
+module System.Taffybar.Widget.Brightness (brightnessBarNew) where
+
+import Control.Concurrent.Chan (dupChan)
+import Graphics.UI.Gtk (Widget)
+import System.Taffybar.Widget.Generic.VerticalBar (BarConfig)
+import System.Taffybar.Widget.Generic.PollingBar (verticalBarFromChannel)
+import System.Taffybar.Information.Brightness (readBrightnessValues)
+
+brightnessBarNew :: BarConfig -> IO Widget
```

Comment:
Yeah, this should just use ChannelWidget.

Also, this starts a new poller every time the widget is created. This means that if someone were to toggle taffybar repeatedly, you would start accumulating channels and threads that are polling this file.

I actually toggle taffybar on one of my monitors quite frequently (on the order of 10 times in the average lifecycle of a taffybar process). If i were to add this widget to my config, I would accumulate a whole bunch of these threads (and they would all be writing to an unused unbounded channel).

There are two ways to fix this. 
- One is to kill the channel/thread whenever the widget is unrealized.
- The better way to handle it is to use the pattern that I've been using for all of my channel widgets that leverages the state mechanism of Taffybar's Context reader to share a single channel for all widgets that want access to the information (This is the purpose of the dupChan btw. it makes the channel a broadcast channel)

You can look at
https://github.com/taffybar/taffybar/blob/a4d757c1509eb44526921c0d90afa157a23db1a7/src/System/Taffybar/Hooks.hs#L22
and
https://github.com/taffybar/taffybar/blob/a4d757c1509eb44526921c0d90afa157a23db1a7/src/System/Taffybar/Information/Battery.hs#L151

for examples of this pattern. Basically, the idea is to use getStateDefault to get the singleton channel instance.

Reply:
It just occured to me that even with the approach that I currently use, there can be a space leak when there are no readers of the channel. We might want to look at using
https://github.com/merijn/broadcast-chan to handle this.

Reply:
I was wondering about the Chan thing - would something like MVar work? Reasoning being, you only care about the current value in the display. If for some reason you've missed a few updates, you don't care about the intermediate / historic values - just the current one.

Reply:
MVar doesn't really give you wakeup behavior. You can just empty the channel whenever you read it. Also when you dupchan you only receive values after the duplication, so unless you are too slow, this isn't going to happen.

Reply:
Oh yeah, takeMVar would only allow one thread to read the value. What about Control.Concurrent.Broadcast, with `listen` and `signal`? Allows multiple consumers to block on a single value, and all get it when a value is published.